src/app/modules/shared-components/components/duration/duration.component.ts

```diff
@@ -34,10 +35,15 @@ export class DurationComponent implements ControlValueAccessor {
       return;
     }
 
-    const [ hours, minutes, seconds ] = value.split(':');
-    this.hours = hours;
-    this.minutes = minutes;
-    this.seconds = seconds;
+    const duration: Duration | null = Duration.fromString(value);
+
+    if (!duration) {
+      return;
+    }
+
+    this.hours = duration.getHours();
+    this.minutes = duration.getMinutes();
+    this.seconds = duration.getSeconds();
```

Comment:
you could create a nice `getHMS` which would make it even cleaner. (assignment on a single line)

Reply:
Fixed